thinc/backends/cupy_ops.py

```diff
@@ -170,39 +170,43 @@ def backprop_clipped_linear(
             )
 
     def backprop_hard_swish(self, dY, X, inplace: bool = False):
-        if X.dtype == "float32" and dY.dtype == "float32":
+        if X.dtype == dY.dtype and X.dtype in ("float32", "float64"):
             return _custom_kernels.backprop_hard_swish(dY, X, inplace=inplace)
         else:
             return super().backprop_hard_swish(dY, X, inplace=inplace)
 
     def backprop_hard_swish_mobilenet(self, dY, X, inplace: bool = False):
-        if X.dtype == "float32" and dY.dtype == "float32":
+        if X.dtype == dY.dtype and X.dtype in ("float32", "float64"):
             return _custom_kernels.backprop_hard_swish_mobilenet(dY, X, inplace=inplace)
         else:
             return super().backprop_hard_swish_mobilenet(dY, X, inplace=inplace)
 
     def mish(self, X, threshold=20.0, inplace=False):
-        if X.dtype == "float32":
+        if X.dtype in ("float32", "float64") and not inplace:
             return _custom_kernels.mish(X, inplace=inplace, threshold=threshold)
         else:
             return super().mish(X, threshold, inplace)
 
     def backprop_mish(self, dY, X, threshold=20.0, inplace=False):
-        if dY.dtype == "float32" and X.dtype == "float32":
+        if X.dtype == dY.dtype and X.dtype in ("float32", "float64"):
             return _custom_kernels.backprop_mish(
                 dY, X, inplace=inplace, threshold=threshold
             )
         else:
             return super().backprop_mish(dY, X, threshold, inplace)
 
     def swish(self, X, inplace=False):
-        if X.dtype == "float32":
+        if X.dtype in ("float32", "float64"):
             return _custom_kernels.swish(X, inplace=inplace, threshold=17.0)
         else:
             return super().swish(X, inplace=inplace)
 
     def backprop_swish(self, dY, X, Y, inplace=False):
-        if X.dtype == "float32" and dY.dtype == "float32":
+        if (
+            X.dtype == dY.dtype
+            and X.dtype == Y.dtype
```

Comment:
```suggestion
            X.dtype == dY.dtype == Y.dtype
```
This works too, right?

Reply:
Nice, I didn't know that! Fixed.